command_line/refine.py

```diff
@@ -383,6 +384,13 @@ def run(args=None, phil=working_phil):
             "tasks.\n"
         )
 
+    if params.refinement.parameterisation.scan_varying is not False:
+        # duplicate crystal if necessary for scan varying
+        if len(experiments) > len(experiments.crystals()):
```

Comment:
This condition is really being used as a proxy for detecting when a crystal is associated with more than one scan. It may be better to actually test for that, as otherwise this behaviour will also be triggered in situations where you don't want to do this. These situations are mostly hypothetical, but not totally unreasonable. Two detectors for one scan is an example - the two-EIGERs-in-a-V-shape, or a separate high angle detector. The code that includes the `Sorry` message mentioned in #994 already does this.

Reply:
@dagewa the correct test is therefore to look to see if #scans > #crystals and split according to this? Since we don't have test coverage for more esoteric combinations (such as I am trying to add support for) I don't know the space of "allowed solutions" - though your 2 detector at the same time combination makes sense.

Reply:
I think I resolved this in the most recent commit - 

```
        # duplicate crystal if necessary for scan varying - will need
        # to compare the scans with crystals - if not 1:1 will need to
        # split the crystals

        crystal_has_scan = {}
        for j, e in enumerate(experiments):
            if e.crystal in crystal_has_scan:
                if e.scan is not crystal_has_scan[e.crystal]:
                    logger.info("Separating crystal model for experiment %d" % j)
                    e.crystal = copy.deepcopy(e.crystal)
            else:
                crystal_has_scan[e.crystal] = e.scan
```

This _should_ duplicate the crystal iff > 1 scan associated with that crystal.

Reply:
That looks good to me :+1: